hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/store/driver/impl/StateStoreFileBaseImpl.java

```diff
@@ -400,10 +403,8 @@ public <T extends BaseRecord> StateStoreOperationResult putAll(
           toWrite.put(recordPath, record);
         } else if (errorIfExists) {
           LOG.error("Attempt to insert record {} that already exists", recordPath);
-          if (metrics != null) {
-            metrics.addFailure(monotonicNow() - start);
```

Comment:
We will have metrics, at the end

Reply:
```
    long end = monotonicNow();
    if (metrics != null) {
      if (success.get()) {
        metrics.addWrite(end - start);
      } else {
        metrics.addFailure(end - start);
      }
    }
    return new StateStoreOperationResult(failedRecordsKeys, success.get());
```

Reply:
without this patch, it used to make sense keeping metric update logic here, because we were returning directly from here. but now we don't need this because metric update is going to take place at the end anyways (just like other cases).